website/docs/reference/appsmith-framework/widget-actions/post-message.md

```diff
@@ -1,11 +1,13 @@
-# Post Message
+# Cross-origin communication
+
+## Post Message
 
 The **Post message** action uses the `postWindowMessage()` function to enable safe cross-origin communication between different [Window](https://developer.mozilla.org/en-US/docs/Web/API/Window) objects; for example, between an app and an iframe embedded within it. This may be useful in situations like:
```

Comment:
Here's what I think should be the structure:
# Cross-origin Communication
   Introduction about the topic and what users can expect in the below section.
## Methods
Give a general overview of methods that are used in cross-origin communication
### postWindowMessage()
Add the existing content to it.
### subscribeParentMessages
Add details about this method
### unsubscribeParentMessages
Add details about this method
## Types of Communication
The existing content continues
### Appsmith app embedded in parent app
existing content
#### From Appsmith to parent app
existing content
#### From parent app to Appsmith
Add examples for this communication using subscribeParentMessages and unsubscribeParentMessages methods.

Reply:
Agree. Slight modification


# Cross-origin Communication
Introduction about the topic and what users can expect in the below section.

## Methods
Give a general overview of methods that are used in cross-origin communication

### postWindowMessage()
Add the existing content to it.

### subscribeParentMessages()
Add details about this method

### unsubscribeParentMessages()
Add details about this method

## Types of Communication
The existing content continues

### Iframe widget in Appsmith app
existing content

#### From Appsmith to embedded page
existing content

#### From embedded page to Iframe widget
existing content

#### From Iframe widget to Appsmith (We missed this piece. we have to check with Anand to confirm whether this is now possible)
Add examples for this communication using subscribeParentMessages and unsubscribeParentMessages methods.

### Appsmith app embedded in parent app
existing content

#### From Appsmith to parent app
existing content

#### From parent app to Appsmith
Add examples for this communication using subscribeParentMessages and unsubscribeParentMessages methods.

Reply:
@eco-monk can you confirm if with the new listeners, we can now also communicate from the Iframe widget inside Appsmith to the Appsmith app window?


@jnikhila @Pranay105 If this is the case, we will have a new section called **From Iframe widget to Appsmith** under the sub-heading **Iframe widget in Appsmith app**

Reply:
I think we can skip the method heading and add it in the intro itself. @danciaclara

Reply:
A very simplified representation of what this method does.
<img width="553" alt="image" src="https://user-images.githubusercontent.com/66776129/212668559-482f8d78-6125-42d9-80e8-7dcca962333e.png">